lib/rex/proto/kerberos/keytab/keytab.rb

```diff
@@ -0,0 +1,151 @@
+    search_prefix :keytab
+
+    # @!attribute [rw] epoch
+    #   @return [Integer]
+    uint32 :epoch
+
+    def get
+      Time.at(epoch)
+    end
+
+    def set(v)
+      self.epoch = v.to_i
+    end
+  end
+
+  class KeytabEntry < BinData::Record
+    endian :big
+    search_prefix :keytab
+
+    # @return [Integer] The number of bytes for the len field
+    LEN_FIELD_BYTE_SIZE = 4
+
+    # @!attribute [rw] len
+    #   @return [Integer] The number of remaining bytes for this record. The length does not include the 4 bytes for this field
+    int32 :len,
+          value: -> {
+            size = [
+              count_of_components,
+              realm,
+              components,
+              name_type,
+              timestamp,
+              vno8,
+              keyblock,
+              # TODO: Omit vno and flags to ensure we generate MIT output for cross-compatibility with ktpass and wireshark
+              vno,
+              flags
+            ].sum { |field| field.to_binary_s.bytes.count }
+
+            size
+         }
+
+    # @!attribute [rw] count_of_components
+    #   @return [Integer]
+    uint16 :count_of_components, value: -> { components.length }
+
+    # @!attribute [rw] realm#
+    #   @return [CountedOctetString]
+    counted_octet_string :realm
+
+    # @!attribute [rw] components
+    #   @return [Array<CountedOctetString>] The components in the principal name, which can be joined by slashes
+    #     to represent the SPN
+    array :components, initial_length: :count_of_components, type: :counted_octet_string
+
+    # @!attribute [rw] name_type
+    #   @return [Integer]
+    # @see Rex::Proto::Kerberos::Model::NameType
+    uint32 :name_type
+
+    # @!attribute [rw] timestamp
+    #   @return [Integer] The time the key entry was created; Can be 0 for keytabs generated by ktpass
+    epoch :timestamp
+
+    # @!attribute [rw] vno8
+    #   @return [Integer] The lower 8 bits of the version number of the key
+    uint8 :vno8
+
+    # @!attribute [rw] keyblock
+    #   @return [KeytabKeyBlock]
+    keyblock :keyblock
+
+    # @!attribute [rw] vno
+    #   @return [Integer]
+    uint32 :vno,
+           initial_value: -> { vno8.to_i },
+           # only present if >= 4 bytes left in entry
+           onlyif: -> { ((len + LEN_FIELD_BYTE_SIZE) - vno.rel_offset) >= 4 }
+
+    # @!attribute [rw] flags
+    #   @return [Integer]
+    uint32 :flags,
+           # only present if >= 4 bytes left in entry
+           onlyif: -> { ((len + LEN_FIELD_BYTE_SIZE) - flags.rel_offset) >= 4 }
+
+    # @return [String] The principal associated with this key tab entry
+    def principal
+      "#{components.to_a.join('/')}@#{realm}"
+    end
+  end
+
+  # Definition from:
+  # https://web.mit.edu/kerberos/krb5-devel/doc/basic/keytab_def.html
+  # http://www.ioplex.com/utilities/keytab.txt
+  # http://web.mit.edu/freebsd/head/crypto/heimdal/doc/doxyout/krb5/html/krb5_fileformats.html
+  #
+  class Keytab < BinData::Record
```

Comment:
```suggestion
  class KRB5::Keytab < BinData::Record
```

Also consider adding comments about how zeitwerk loading can impact a need to combine classes in a single module file.